src/main/scala/com/netflix/edda/mongo/MongoDatastore.scala

```diff
@@ -69,18 +69,27 @@ object MongoDatastore {
     import collection.JavaConverters._
     obj match {
       case o: BasicDBObject => {
-        o.keySet.asScala.map((key: String) => (key -> mongoToScala(o.get(key)))).toMap
+        o.keySet.asScala.map((key: String) => (mongoDecodeString(key) -> mongoToScala(o.get(key)))).toMap
       }
       case o: BasicDBList => {
         List.empty[Any] ++ o.asScala.map(mongoToScala(_))
       }
       case o: Date => new DateTime(o)
+      case o: String => mongoDecodeString(o)
       case o: AnyRef => o
       case null => null
       case other => throw new java.lang.RuntimeException("mongoToScala: don't know how to handle: " + other)
     }
   }
 
+  def mongoDecodeString(str: String): String = {
+    str.replace("\\uFF0E", ".").replace("\\uFF04", "$")
+  }
+
+  def mongoEncodeString(str: String): String = {
+    str.replaceAll("[\\.]", "\\\\uFF0E").replaceAll("[\\$]", "\\\\uFF04")
```

Comment:
Any reason this doesn't use `replace` like `mongoDecodeString`? I think it is easier to read without the regex escaping:

```scala
str.replace(".", "\\uFF0E").replace("$", "\\uFF04")
```